src/content.ts

```diff
@@ -0,0 +1,53 @@
+import { getFromLocalStorage } from "./chrome-utils/localStorage";
+
+const localStorageKey = "isExtensionEnabled"; // BAD! maintain keys centrally.
+
+document.addEventListener('mouseup', async function (event) {
+    const isEnabled = await getFromLocalStorage(localStorageKey) as boolean | undefined;
+    const selectedText = window.getSelection()?.toString().trim() ?? "";
+    if (isEnabled !== true) {
+        return;
+    }
```

Comment:
This file is almost a clean copy/paste. Only added 3-4 lines here to return when the plugin is disabled.